### PR TITLE
add user stories for disk support

### PR DESCRIPTION
**Pull request by kannon92**

### CODE REVIEW

File: enhancements/machine-api/multiple-disk-support.md [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+### API Extensions
+
+### Implementation Details/Notes/Constraints
+
+### Topology Considerations
+
+#### Hypershift / Hosted Control Planes
+
+This proposal does not affect HyperShift.
+HyperShift does not leverage Machine API.
```
Comment (derekwaynecarr) [inline enhancements/machine-api/multiple-disk-support.md:106]: Hosted Control Planes do support the ability to inject a MachineConfig into the NodePool definition.  For the use cases described above, it seems like we should be able to provide the same capability for either form factor, and customers (particularly the swap scenario) would benefit ensuring we have coverage for both form factors.

Comment (derekwaynecarr) [reply enhancements/machine-api/multiple-disk-support.md:106]: @sjenning @mrunalp the use cases oriented around swap are valuable in the HCP scenario, wanted to make sure we ensure that we can have a consistent set of disk layout options even if in the HCP case it is defined externally via NodePool/MachineConfig injection.

File: enhancements/machine-api/multiple-disk-support.md [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+Custormers request the ability to add disks for day 0 and day 1 operations. Some of the common areas include designed disk for etcd, dedicated disk for swap partitions, container runtime filesystem, and a separate filesystem for container images.
+
+All of these features are possible to support through a combination of machine configs and machine API changes.
+However, the support varies across the use cases and there is a need to define a general API for disk support so we can have a common experience across different methods.
+
+### Workflow Description
+
+### Goals
+
+TBD
```
Comment (mdbooth) [inline enhancements/machine-api/multiple-disk-support.md:48]: * Define a common interface for infrastructure platforms to implement to use additional disks for a defined set of specific uses
* Implement common behaviour to safely use the above disks when they have been presented by the infrastructure platform

File: enhancements/machine-api/multiple-disk-support.md (hunk 1 above)
Comment (kannon92) [reply enhancements/machine-api/multiple-disk-support.md:106]: @derekwaynecarr what would be the motivation for HCP swap? We were thinking that swap should only be enabled on the worker nodes. I don't see a benefit of swap on control plane nodes at the moment.

Comment (kannon92) [reply enhancements/machine-api/multiple-disk-support.md:106]: > Hosted Control Planes do support the ability to inject a MachineConfig into the NodePool definition on the management cluster. For the use cases described above, it seems like we should be able to provide the same capability for either form factor, and customers (particularly the swap scenario) would benefit ensuring we have coverage for both form factors.

I should look more into this. Do you have a link or a design doc handy for this?

Comment (derekwaynecarr) [reply enhancements/machine-api/multiple-disk-support.md:170]: The motivation is to ensure that you can enable the desired disk layout for swap on the worker nodes that join a Hosted Control Plane.  Those worker nodes are configured via the NodePool abstraction on the management cluster which supports the ability to inject a MachineConfig.

For customers exploring OpenShift Virtualization to support their virtualization workload, we see a lot of interest in having OpenShift Virtualization running on a cluster that uses the HCP form-factor with bare-metal workers in order to bring the number of physical nodes needed to support the number of control planes down.  This is important if you have a large number of virtualization machines and therefore need multiple clusters to support virtualization in a given data center.

Comment (derekwaynecarr) [reply enhancements/machine-api/multiple-disk-support.md:170]: See the example documented here:
https://hypershift-docs.netlify.app/how-to/automated-machine-management/configure-machines/

Comment (kannon92) [reply enhancements/machine-api/multiple-disk-support.md:170]: Thank you for the link. We will make sure to consider NodePool in this design.

**Pull request closed without merging**